apis/ecs/v1alpha1/custom_types.go

```diff
@@ -0,0 +1,222 @@
+	// target type, not instance, because tasks that use the awsvpc network mode
+	// are associated with an elastic network interface, not an Amazon EC2 instance.
+	LoadBalancers []*CustomLoadBalancer `json:"loadBalancers,omitempty"`
+
+	// The network configuration for the service. This parameter is required for
+	// task definitions that use the awsvpc network mode to receive their own elastic
+	// network interface, and it is not supported for other network modes. For more
+	// information, see Task networking (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-networking.html)
+	// in the Amazon Elastic Container Service Developer Guide.
+	NetworkConfiguration *CustomNetworkConfiguration `json:"networkConfiguration,omitempty"`
+
+	// The family and revision (family:revision) or full ARN of the task definition
+	// to run in your service. If a revision is not specified, the latest ACTIVE
+	// revision is used.
+	//
+	// A task definition must be specified if the service is using either the ECS
+	// or CODE_DEPLOY deployment controllers.
+	// +optional
+	// +crossplane:generate:reference:type=TaskDefinition
+	TaskDefinition         *string         `json:"taskDefinition,omitempty"`
+	TaskDefinitionRef      *xpv1.Reference `json:"taskDefinitionRef,omitempty"`
+	TaskDefinitionSelector *xpv1.Selector  `json:"taskDefinitionSelector,omitempty"`
+}
+
+// CustomEFSAuthorizationConfig provides custom parameters for the
+// EFSAuthorizationConfig type
+type CustomEFSAuthorizationConfig struct {
+	// The Amazon EFS access point ID to use. If an access point is specified, the
+	// root directory value specified in the EFSVolumeConfiguration must either
+	// be omitted or set to / which will enforce the path set on the EFS access
+	// point. If an access point is used, transit encryption must be enabled in
+	// the EFSVolumeConfiguration. For more information, see Working with Amazon
+	// EFS Access Points (https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html)
+	// in the Amazon Elastic File System User Guide.
+	// +optional
+	// +crossplane:generate:reference:type=github.com/crossplane-contrib/provider-aws/apis/efs/v1alpha1.AccessPoint
+	AccessPointID         *string         `json:"accessPointID,omitempty"`
+	AccessPointIDRef      *xpv1.Reference `json:"accessPointIDRef,omitempty"`
+	AccessPointIDSelector *xpv1.Selector  `json:"accessPointIDSelector,omitempty"`
+
+	// Determines whether to use the Amazon ECS task IAM role defined in a task
+	// definition when mounting the Amazon EFS file system. If enabled, transit
+	// encryption must be enabled in the EFSVolumeConfiguration. If this parameter
+	// is omitted, the default value of DISABLED is used. For more information,
+	// see Using Amazon EFS Access Points (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/efs-volumes.html#efs-volume-accesspoints)
+	// in the Amazon Elastic Container Service Developer Guide.
+	IAM *string `json:"iam,omitempty"`
```

Comment:
think we can use here Ref/Selector ?

Reply:
I think IAM is an example of the inconsistent AWS APIs, from the api documentation (https://docs.aws.amazon.com/sdk-for-go/api/service/ecs/), it looks like this is effectively a boolean flag, with an enum value of "ENABLED" or "DISABLED", rather than a pointer to an IAM resource:
```
const (
    // EFSAuthorizationConfigIAMEnabled is a EFSAuthorizationConfigIAM enum value
    EFSAuthorizationConfigIAMEnabled = "ENABLED"

    // EFSAuthorizationConfigIAMDisabled is a EFSAuthorizationConfigIAM enum value
    EFSAuthorizationConfigIAMDisabled = "DISABLED"
)
```